### PR TITLE
Android shared cache

### PR DESCRIPTION
**Pull request by novalisdenahi**
### Description

Add ConfigCatPreferencesCache description to Android

### Requirement checklist

- [x] I have validated my changes on a test/local environment.
- [ ] I have checked the SNYK/Dependabot reports and applied the suggested changes.
- [ ] (Optional) I have updated outdated packages.

### CODE REVIEW

File: website/docs/sdk-reference/android.md [hunk 1]
```diff
@@ -476,6 +476,17 @@ ConfigCatClient client = ConfigCatClient.get("#YOUR-SDK-KEY#", options -> {
     options.cache(new MyCustomCache()); // inject your custom cache
 });
 ```
+### Platform specific cache
+
+The SDK provides the `ConfigCatPreferencesCache` as platform specific cache. The cache uses the Android `SharedPreferences` to store the downloaded `config.json`.
+
+It has a dependency on `android.content.Context`, so it won't be enabled by default, but it can be explicitly set by providing an appropriate `Context`.
```
Comment (sigewuzhere) [inline website/docs/sdk-reference/android.md:483]: This sentence has 3 'it's. I don't know which refers to what. I suggest rephrasing.

**Review by sigewuzhere — CHANGES_REQUESTED** [review]

File: website/docs/sdk-reference/android.md [hunk 2]
```diff
@@ -476,6 +476,17 @@ ConfigCatClient client = ConfigCatClient.get("#YOUR-SDK-KEY#", options -> {
     options.cache(new MyCustomCache()); // inject your custom cache
 });
 ```
+### Platform specific cache
```
Comment (z4kn4fein) [inline website/docs/sdk-reference/android.md:479]: Could you please format this similar to the kotlin one, like:
```
## Cache
<Details about shared pref cache>

### Custom Cache
...
```

**Review by z4kn4fein — CHANGES_REQUESTED** [review]

**Review by z4kn4fein — CHANGES_REQUESTED** [review]

File: website/docs/sdk-reference/android.md (hunk 1 above)
Comment (novalisdenahi) [reply website/docs/sdk-reference/android.md]: Rephrased.

File: website/docs/sdk-reference/android.md (hunk 2 above)
Comment (novalisdenahi) [reply website/docs/sdk-reference/android.md]: Reworked.

**Review by z4kn4fein — CHANGES_REQUESTED** [review]

**Review by z4kn4fein — DISMISSED** [review]

**Review by z4kn4fein — APPROVED** [review]

**Pull request merged**